adoc/chapters/programming_interface.adoc

```diff
@@ -1800,8 +1800,14 @@ info::device::max_work_group_size
 ----
 
     @ [.code]#size_t#
-   a@ Returns the maximum number of work-items that are permitted in a work-group executing a kernel on a single compute unit.
+   a@ Returns the maximum number of work-items that this device is capable of
+      executing in a work-group.
       The minimum value is 1.
+      This value is an upper limit and will not necessarily maximize
+      performance.
+      Some kernels may not be able to execute this many work-items in a
```

Comment:
Perhaps we should rephrase this to be more instructive as to why some kernels may not be able to execute the maximum number of work-items. Something like:

"The actual maximum number of work-items that a kernel can execute in a work-group will depend on other factors of the kernel which can impact occupancy.

Reply:
I'm not sure we want the spec to speculate about why a kernel might have a lower limit.  The reasons for the lower limit seem like implementation details.

Reply:
I agree, though I think we can provide a bit more context without being too specific, perhaps we could say that the actual maximum number of work-items per work-group for a compiled kernel is implementation defined, and may not reach the maximum specific here.

Reply:
Makes sense.  See ed18871275cb6911e52c8100141996e73e76c7f8